org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/WorkspaceEventsHandler.java

```diff
@@ -66,6 +90,22 @@ void didChangeWatchedFiles(DidChangeWatchedFilesParams param){
 		}
 	}
 
+	private ICompilationUnit createCompilationUnit(ICompilationUnit unit) {
+		try {
+			unit.getResource().refreshLocal(IResource.DEPTH_ONE, new NullProgressMonitor());
+			if (unit.getResource().exists()) {

```

Comment:
shouldn't we check the resource exists BEFORE refreshing it?

Reply:
this method is largely similar to the block you modified in DocumentLifeCycleHandler.handleOpen, yet the conditions are different. Here the resource needs to exists, in handleOpen, it must not be accessible (doesn't exist). Seems dubious. Both places should probably be refactored to call the same logic

Reply:
The resource must exist after refreshing.